Bluetooth: btusb: Improve support for Actions Semi ATS2851 based devices

EzPzDev opened the pull request:
Add two more quirks to resume the device initialization and basic operation as the device seems not to support "Read Transmit Power" and "Set Extended Scan Parameters".

EzPzDev committed:
Bluetooth: btusb: Improve support for Actions Semi ATS2851 based devices

Add two more quirks to resume the device initialization and basic
operation as the device seems not to support "Read Transmit Power"
and "Set Extended Scan Parameters".

pelwell commented:
@holtmann You added the HCI_QUIRK_BROKEN_ERR_DATA_REPORTING quirk for the ATS2851. Does also setting HCI_QUIRK_BROKEN_READ_TRANSMIT_POWER and HCI_QUIRK_BROKEN_EXT_SCAN sound reasonable to you?

EzPzDev committed:
Add new quirk for broken set random RPA timeout for ATS2851

The ATS2851 based controller advertises support for command "LE Set Random
Private Address Timeout" but does not actually implement it, impeding the
controller initialization.

EzPzDev commented:
Sorry, this was originally from upstream. I forgot to add Co-authored-by: Raul Cheleguini.

pelwell commented:
Unfortunately what you've done is to remove the useful commit message and delay the approval process. We are (almost) always happy to accept back-ports from newer kernels because it means i) that they have been approved, and ii) that the minimal support burden disappears once we reach that kernel version.

In future, cherry-pick the back-ports, adding a line at the start of the commit message saying "commit <hash> upstream" but otherwise leaving them unchanged, Alternatively, create an Issue asking us to back-port them.

pelwell commented:
See https://github.com/raspberrypi/linux/pull/5465 for how it should be done.
